src/_ert/forward_model_runner/forward_model_step.py

```diff
@@ -203,8 +204,19 @@ def _run(self) -> Generator[Start | Exited | Running]:
                     yield potential_exited_msg
                     return
 
-            (memory_rss, cpu_seconds, oom_score, pids) = _get_processtree_data(process)
-            max_cpu_seconds = max(max_cpu_seconds, cpu_seconds or 0)
+            (memory_rss, cpu_seconds_snapshot, oom_score, pids) = _get_processtree_data(
+                process
+            )
+            for pid, seconds in cpu_seconds_snapshot.items():
```

Comment:
You could extract this part to make it easier to test:
```python
@dataclass
class CpuTimer:
  _cpu_seconds_pr_pid: dict[str, float] = field(default_factory=dict, init=False)

  def update(self, cpu_seconds_snapshot):
    ...

  def total_cpu_seconds(self):
    return sum(self._cpu_seconds_pr_pid.values())
```

Then you can easily test the logic by giving it fake `cpu_seconds_snapshots`.